### PR TITLE
Add codecov support

### PR DESCRIPTION
**Pull request by ashulga-r7**
_in progress_

https://github.com/seamusc/lepy/projects/1#card-17369260

Add code coverage analysis:
 - based on Codecov https://docs.codecov.io/docs
 - integrated with Travic CI

@scawley-r7 @seamusc @mzuber-r7 @icakmak-r7

### CODE REVIEW

**Comment by mzuber-r7** [comment]
does it you make sense to make ```tools/run_coverage.sh``` ...?

**Review by mzuber-r7 — APPROVED** [review]

**Comment by ashulga-r7** [comment]
> does it you make sense to make `tools/run_coverage.sh` ...?

ok will do it for consistency with `./tools/pylint.sh`

File: .codecov.ymll [hunk 1]
```diff
@@ -0,0 +1,26 @@
+codecov:
+  notify:
+    require_ci_to_pass: yes
+
+coverage:
+  precision: 2
+  round: down
+  range: "70...100"
```
Comment (ashulga-r7) [inline .codecov.ymll]: Guess this one is for discussion

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    require_ci_to_pass: yes
+
+coverage:
+  precision: 2
+  round: down
+  range: "70...100"
+
+  status:
+    project: yes
+      target: 20%
```
Comment (ashulga-r7) [inline .codecov.yml:13]: Defining targets for code coverage: https://docs.codecov.io/docs/commit-status
Current value is super low in order to allow existing PRs.
We should increase targets in the future.

**Pull request closed without merging**